cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/DynamicTransactionExecutor.java

```diff
@@ -118,7 +118,16 @@ public Void apply(Void input) throws Exception {
   }
 
   private <I, O> O executeOnce(Function<I, O> function, I input) throws TransactionFailureException {
-    TransactionContext txContext = txContextSupplier.get();
+    TransactionContext txContext;
+    try {
+      txContext = txContextSupplier.get();
+    } catch (RuntimeException e) {
```

Comment:
supplier.get() cannot throw checked exceptions. Therefore, the context supplier must wrap a possible TransactionFailureException into a runtime exception.

this can happen, for example, if the supplier has an active transaction, and this would represent a nested transaction.

I added this as a comment to the code